tensorflow/core/common_runtime/pluggable_device/BUILD

```diff
@@ -0,0 +1,155 @@
+load(
+    "//tensorflow:tensorflow.bzl",
+    "tf_copts",
+)
+load(
+    "//tensorflow/core/platform:rules_cc.bzl",
+    "cc_library",
+)
+
+package(
+    default_visibility = [
+        "//tensorflow:internal",
+    ],
+    licenses = ["notice"],  # Apache 2.0
+)
+
+filegroup(
+    name = "pluggable_device_runtime_headers",
+    srcs = [
+        "pluggable_device.h",
+        "pluggable_device_bfc_allocator.h",
+        "pluggable_device_context.h",
+        "pluggable_device_factory.h",
+        "pluggable_device_init.h",
+        "pluggable_device_process_state.h",
+        "pluggable_device_util.h",
+        "//tensorflow/core/common_runtime/device:device_runtime_headers",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "pluggable_device_runtime_impl",
+    srcs = [
+        "pluggable_device.cc",
+        "pluggable_device_context.cc",
+        "pluggable_device_factory.cc",
+        "pluggable_device_process_state.cc",
+        "pluggable_device_util.cc",
+    ],
+    hdrs = [":pluggable_device_runtime_headers"],
+    copts = tf_copts(),
+    deps = [
+        ":pluggable_device_bfc_allocator",
+        ":pluggable_device_init_impl",
+        "//tensorflow/core:core_cpu_lib",
+        "//tensorflow/core:framework",
+        "//tensorflow/core:framework_internal",
+        "//tensorflow/core:graph",
+        "//tensorflow/core:lib",
+        "//tensorflow/core:lib_internal",
+        "//tensorflow/core:protos_all_cc",
+        "//tensorflow/core/platform:stream_executor",
+        "//tensorflow/core/common_runtime/device:device_event_mgr",
+        "//tensorflow/core/platform:tensor_float_32_utils",
+    ],
+    alwayslink = 1,
+)
+
+cc_library(
+    name = "pluggable_device_plugin_init",
+    srcs = [
+        "pluggable_device_plugin_init.cc",
+    ],
+    hdrs = [
+        "pluggable_device_plugin_init.h",
+        ":pluggable_device_runtime_headers",
+        "//tensorflow/c:headers",
+        "//tensorflow/core/common_runtime:core_cpu_base_headers",
+        "//tensorflow/core/public:session_options.h",
```

Comment:
The `hdrs` section [lists the headers that this module exports](https://docs.bazel.build/versions/master/be/c-cpp.html#cc_library.hdrs). (The header files that its dependency will be able to use.) 

Why does PluggableDevice need to export headers of other targets?